cloudbuild.yaml

```diff
@@ -1,17 +1,42 @@
-# This file is used to build and deploy the app into a GKE cluster using Google
-# Cloud Build.
+# Cloudbuild.yaml to deploy to staging
 #
-# Requirements:
-# - Give the Google Cloud Build service account "Kubernetes Engine Developer"
-#   IAM role on the GCP project.
-# - Set up Google Cloud Build trigger on Cloud Console.
+# PREREQUISITES:
+# - Cloud Build service account must have roles: "Kubernetes Engine Developer", "Storage Object Creator", \
+#   "Storage Object"
 
 steps:
-- name: gcr.io/k8s-skaffold/skaffold:v0.16.0
-  args: ['skaffold', 'run', '-f=skaffold.yaml']
+
+# authenticate to GKE cluster
+- id: 'Get GKE credentials'
+  name: gcr.io/cloud-builders/gcloud
+  args: ['container','clusters','get-credentials','demo-app-staging']
   env:
   - 'CLOUDSDK_COMPUTE_ZONE=us-central1-b'
+
+# make latest version of the Skaffold community builder available in this project
+- id: 'clone community builders repo'
+  name: gcr.io/cloud-builders/git
+  args: ['clone','https://github.com/GoogleCloudPlatform/cloud-builders-community']
+
+- id: 'build Skaffold builder'
+  name: gcr.io/cloud-builders/docker
+  args: 
+    [
+      'build',
+      '--tag', 'gcr.io/$PROJECT_ID/skaffold',
```

Comment:
you can call this skaffold for simplicity, we aren't pushing this image.

Reply:
good call; changed.